### ISSUE TITLE
Incorrect handling of EO decision for PETPT

### ISSUE THREAD

**Issue by pauldhein** [ml4ai/delphi#206]
We have a few issues left to handle for the `__decision__` nodes for `EO`.  First it appears that we have a variable name mismatch as captured in this view of the GrFN output from PETPT:

<img width="1104" alt="Screen Shot 2019-03-08 at 3 57 34 PM" src="https://user-images.githubusercontent.com/14334973/54055403-fd3f0900-41ba-11e9-9c48-7269d8baf421.png">

1. The output of a decision node should be a setting of a particular value (for now, might need to be a function in the future). Therefore I would like the name of the output of `petpt__decision__eo_3` to be changed from `decision__eo_3` to `eo_3`. 

2. Some incorrect wiring seems to be occurring in the lambdas:
```python
def petpt__assign__eo_0(eeq):
    return (eeq*1.1)

def petpt__condition__IF_2_0(tmax):
    return (tmax>35.0)

def petpt__assign__eo_1(eeq, tmax):
    return (eeq*(((tmax-35.0)*0.05)+1.1))

def petpt__condition__IF_3_0(tmax):
    return (tmax<5.0)

def petpt__assign__eo_2(eeq, tmax):
    return ((eeq*0.01)*math.exp((0.18*(tmax+20.0))))

def petpt__decision__eo_3(IF_3_0, eo_0, eo_2):
    return eo_2 if IF_3_0 else eo_0

def petpt__decision__eo_4(IF_2_0, eo_1, decision__eo_3):
    return decision__eo_3 if IF_2_0 else eo_1
```
By these declarations I'd expect `petpt__decision__eo_3` to take in `IF_2_0, eo_0, eo_1` as variables, not `IF_3_0, eo_0, eo_2`. Second `petpt__decision__eo_4` should take `IF_3_0, eo_2, eo_3` as variables and the function body should be:
```python
return eo_2 if IF_3_0 else eo_3
```
Since we expect to return the `ELSE IF` value if our condition evaluates to true and we want `eo_3` to remain the value for `EO` otherwise.

@pratikbhd how long do you expect it will take for you to make these changes? Can they be done by the end of the day?

**Comment by pratikbhd** [comment]
I just pushed the changes for point 1 above on branch `issue_206` and wanted some clarification regarding the second point before I start with any changes.
Here is what the GrFN JSON for the decision blocks of `EO` variable in `PETPT.for` currently looks like:
```
    {
      "name": "petpt__decision__eo_3",
      "type": "decision",
      "target": "eo",
      "reference": 25,
      "sources": [
        {
          "name": "IF_3_0",
          "type": "variable"
        },
        {
          "name": "eo_0",
          "type": "variable"
        },
        {
          "name": "eo_2",
          "type": "variable"
        }
      ]
    },
    {
      "name": "petpt__decision__eo_4",
      "type": "decision",
      "target": "eo",
      "reference": 22,
      "sources": [
        {
          "name": "IF_2_0",
          "type": "variable"
        },
        {
          "name": "eo_3",
          "type": "variable"
        },
        {
          "name": "eo_1",
          "type": "variable"
        }
      ]
    }
```
And here is how the various instances of the `EO` variable being assigned:

1. `EO = EEQ * 1.1` = eo_0
2. `EO = (EEQ * (((TMAX - 35.0) * 0.05) + 1.1))` = eo_1
3. `EO = ((EEQ * 0.01) * EXP((0.18 * (TMAX + 20.0))))` = eo_2

According to our previous discussion, this is what I thought the `IF` decision blocks should look like (correct me if I am wrong here):

```
IF__2_0:
    assign eo_1
ELSE:
    IF__3_0:
        assign eo_2
```
So, `petpt__decision__eo_3` evaluates `IF__3_0` which assigns `eo_3` (the output of `petpt__decision__eo_3`) as either `eo_2` or `eo_0` (since there is no `else` in `IF__3_0`, the value of `eo_3` would fall back to the value of `eo_0`).  

Then, `petpt__decision__eo_4` evaluates `IF__2_0` which assigns `eo_4` (the output of `petpt__decision__eo_4`) as either `eo_1` or `eo_3` (which is the output of the `else` in `IF__2_0`).

This correlates with the output that is being currently produced in the GrFN JSON and the lambda file. 

I don't completely understand how you are defining `petpt__decision__eo_3` and `petpt__decision__eo_4` to say that their inputs should be different than what they currently are so more information about that would be helpful for me to make further changes if need be.
Thanks.

**Comment by pauldhein** [comment]
The problem is that to maintain the order of the original Fortran code `IF_2_0` should be evaluated before `IF_3_0` and the output from the `IF_2_0` __decision__ should be fed into the `IF_3_0` __decision__. The reverse is happening in the lambdas file that you have generated. To answer your comment, it will not make a difference in this case, but it could in other cases. Consider the following word problem:

```
Add 3 to x if y < 3 otherwise if n is greater than 6 add 5 to x.
```

This will generate the following Fortran:
```fortran
IF (y .LT. 3) THEN
  X = X + 3
ELSE IF (N .GT. 6) THEN
  X = X + 5
ENDIF
```

Following the generation strategy of your lambdas the following python would be generated:

```python
if n > 6:
  x += 5
elif y < 3:
  x += 3
```

Now this is a problem, because in this case both `n > 6` and `y < 3` could be true, and your code would have changed the order in which they were evaluated, thus changing which one gets to evaluate to true, and thus incorrectly determining the value of `x`.

Do you see how the generated code is incorrect, and why although it will not matter for this specific problem it will matter for generality?

**Comment by pauldhein** [comment]
Closed by #208
